packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -51,6 +51,24 @@ void main() {
     });
   });
 
+  {
+    bool retried = false;
+    group('group retry flag allows test to run multiple times', () {
+      testWidgets('primary', (WidgetTester tester) async {
+        addTearDown(() => retried = true);
+        expect(retried, isTrue);
+      });
+    }, retry: 1);
+  }
+
+  {
+    bool retried = false;
+    testWidgets('testWidget retry flag allows test to run multiple times', (WidgetTester tester) async {
+      addTearDown(() => retried = true);
+      expect(retried, isTrue);
+    }, retry: 1);
+  }
+
```

Comment:
I think this'd be more readable if you had these in groups instead of just using `{}` to create new scopes. A nested group in the case of the group retry test.

Reply:
done (waiting for CI)